### DISCUSSION TITLE
Add fragmentation strategies

### DISCUSSION THREAD

**Discussion by KasperZutterman** [TREEcg/LDES-Action#89] [Ideas]
Possible fragmentation strategies that can be added:
- chronological:
  - normal
  - reverse
  - day per day (range e.g. per week)
- lexicographical (alphabetical):
  - normal
  - reverse
  - per substring
- geographical
- object version
- filter

**Comment by pietercolpaert** [comment]
We’re working on `bucketizers` which simply fragment an LDES streamingly and add a `ldes:bucket` predicate to the member with one or more buckets this member can be stored at. I would love a generic Bucketizer fragmentation strategy in here, in which you can configure the name of an NPM bucketizer package (such as the [substring bucketizer](https://www.npmjs.com/package/@treecg/ldes-substring-bucketizer)) and the LDES action will work. This way, no new release of the LDES action would be needed for another fragmentation strategy to work :-)

**Comment by KasperZutterman** [comment]
This may be possible using the [Dynamic import() expression](https://nodejs.org/api/esm.html#esm_import_expressions).

**Comment by pietercolpaert** [comment]
I don’t think we’d need dynamic import, as we can just add a step in our workflow to also install the right bucketizer, no?

**Comment by ddvlanck** [comment]
@pietercolpaert : installing a bucketizer package through configuration is not (yet) possible because GitHub Action requires the build result (dist folder) to be pushed onto GitHub. It uses that folder to run the action and the npm packages are already bundled in that folder, and thus restricting a bucketizer to be added at runtime.

**Comment by pietercolpaert** [comment]
Ok, then we need to fallback to just having a bucketizer repo that keeps a package.json with all known bucketizers as dependencies and exports them?

**Comment by ddvlanck** [comment]
We already have a repository with our bucketizers: https://github.com/TREEcg/bucketizers. So, instead of creating another repo, I suggest to add a new package (e.g. `bucketizer-repo`) with that functionality.

**Comment by pietercolpaert** [comment]
Or just `bucketizers`?
